server/data/applications.ts

```diff
@@ -0,0 +1,298 @@
+        const cycle: any = await cycleCollection.findOne({
+            _id: new ObjectId(cycleId)
+        });
+        if(cycle === null) {
+            throw new Error("There is no cycle with that id.");
+        }
+
+        //Grabs all applicatins in cycle 
+        const cycleApps: object[] = cycle["applications"];
+        let retApps: Array<object> = []; 
+        for(let app of cycleApps) {
+            app["_id"] = app["_id"].toString();
+            retApps.push(app);
+        }
+
+        return retApps;
+    },
+
+    /**
+     * Creates a new application within a given cycle
+     * @param {string} cycleId The cycle to add the application to
+     * @param {string} company Company the application is for
+     * @param {string} position Position beign applied to
+     * @param {string} location Location being applied to
+     * @param {number} salary Position salary 
+     * @param {string} jobPostUrl URL the application was found at
+     * @param {string} cardColor Chosen color of application card
+     * @param {string} description Application description
+     * @returns {Promise<object>} Returns the created application object if successful anf throws
+     * and error otherwise
+     */
+    create: async (cycleId: string, company: string, position: string, location: string,
+        salary: number, jobPostUrl: string, cardColor: string, description: string):
+        Promise<object> => {
+        if(!cycleId || !checkObjectId(cycleId)) {
+            throw new Error("A proper cycle id must be provided.");
+        }
+
+        if(!company || !checkNonEmptyString(company)) {
+            throw new Error("A company name must be provided.");
+        }
+
+        if(!position || !checkNonEmptyString(position)) {
+            throw new Error("A position name must be provided.");
+        }
+
+        if(!location || !checkNonEmptyString(location)) {
+            throw new Error("A location must be provided.");
+        }
+
+        if(!salary || !checkPositiveNumber(salary)) {
+            throw new Error("A salary must be provided.");
+        }
+        if(!jobPostUrl || !checkNonEmptyString(jobPostUrl)) {
+            throw new Error("An application url must be provided.");
+        }
+
+        if(!cardColor || !checkNonEmptyString(cardColor)) {
+            throw new Error("A card color must be selected.");
+        }
+
+        if(!description || !checkNonEmptyString(description)) {
+            throw new Error("A description must be provided.");
+        }
+        
+        //Unsure of how the logo is getting stored if we are using that auto logo finder
+        let newApp: object = {
+            _id: new ObjectId(),
+            company: company,
+            position: position,
+            logo: null,
+            location: location,
+            salary: salary, 
+            jobPostUrl: jobPostUrl,
+            cardColor: cardColor,
+            description: description,
+            progress: 0,
```

Comment:
I believe we said that it would be easier to map the int value (0-4) to one of the strings in the progress bar when we display it so that a 0 would mean "In progress" and a 4 would mean "Ghosted". If we think it'll be easier to do a string though it should be an easy swap

Reply:
0 (IP) - 1 (Offer) - 2 (Reject) - 3 (Waitlist) - 4 (TBD)